src/frontends/tensorflow/src/frontend.cpp

```diff
@@ -97,6 +102,9 @@ ov::frontend::InputModel::Ptr FrontEnd::load_impl(const std::vector<ov::Any>& va
         if (GraphIteratorProto::is_supported(model_path)) {
             // handle binary protobuf format
             return std::make_shared<InputModel>(std::make_shared<GraphIteratorProto>(model_path), m_telemetry);
+        } else if (GraphIteratorSavedModel::isSavedModel(model_path)) {
+            auto graph_iterator = std::make_shared<::ov::frontend::tensorflow::GraphIteratorSavedModel>(model_path);
+            return std::make_shared<InputModel>(graph_iterator, m_telemetry, graph_iterator->getVariablesIndex());
```

Comment:
You dont transfer variables index between objects in your suggestion.
Otherwise I’ll need to transfer knowledge about exact graphiterator into the InputModel class.
It doesnt look like a good idea.

Reply:
yes, please transfer variable. I missed it